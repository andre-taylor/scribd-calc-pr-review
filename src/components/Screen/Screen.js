import React from "react"
import PropTypes from "prop-types"

import styles from "./Screen.module.css"

const Screen = ({ screenValue }) => {
  return <output className={styles.screen}>{screenValue}</output>
}

export default Screen

Screen.propTypes = {
  screenValue: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired
}
