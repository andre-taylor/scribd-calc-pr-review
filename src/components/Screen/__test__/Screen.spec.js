import React from "react"
import renderer from "react-test-renderer"

import Screen from "../Screen"

describe("<Screen />", () => {
  it("renders correctly with required props as a string", () => {
    const component = renderer.create(<Screen screenValue={"1234"} />)
    expect(component).toMatchSnapshot()
  })
  it("renders correctly with required props as a number", () => {
    const component = renderer.create(<Screen screenValue={1234} />)
    expect(component).toMatchSnapshot()
  })
})
