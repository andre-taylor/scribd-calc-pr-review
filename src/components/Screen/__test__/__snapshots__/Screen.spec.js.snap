// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Screen /> renders correctly with required props as a number 1`] = `
<output
  className="screen"
>
  1234
</output>
`;

exports[`<Screen /> renders correctly with required props as a string 1`] = `
<output
  className="screen"
>
  1234
</output>
`;
